cmd/repo-updater/repos/bitbucketserver.go

```diff
@@ -318,61 +325,136 @@ func newBitbucketServerConnection(config *schema.BitbucketServerConnection, cf h
 		return nil, err
 	}
 
+	exclude := make(map[string]bool, len(config.Exclude))
+	for _, r := range config.Exclude {
+		if r.Name != "" {
+			exclude[strings.ToLower(r.Name)] = true
+		}
+
+		if r.Id != 0 {
+			exclude[strconv.Itoa(r.Id)] = true
+		}
+	}
+
 	client := bitbucketserver.NewClient(baseURL, cli)
 	client.Token = config.Token
 	client.Username = config.Username
 	client.Password = config.Password
 
 	return &bitbucketServerConnection{
-		config: config,
-		client: client,
+		config:  config,
+		exclude: exclude,
+		client:  client,
 	}, nil
 }
 
 type bitbucketServerConnection struct {
-	config *schema.BitbucketServerConnection
-	client *bitbucketserver.Client
+	config  *schema.BitbucketServerConnection
+	exclude map[string]bool
+	client  *bitbucketserver.Client
 }
 
-func (c *bitbucketServerConnection) listAllRepos(ctx context.Context) <-chan *bitbucketserver.Repo {
-	perPage := 100
-	ch := make(chan *bitbucketserver.Repo, perPage)
+func (c *bitbucketServerConnection) excludes(r *bitbucketserver.Repo) bool {
+	name := r.Slug
+	if r.Project != nil {
+		name = r.Project.Key + "/" + name
+	}
+	return c.exclude[strings.ToLower(name)] ||
+		c.exclude[strconv.Itoa(r.ID)] ||
+		c.config.ExcludePersonalRepositories && r.IsPersonalRepository()
```

Comment:
I'm guessing the evaluation rules work out here, but this makes me uneasy. Can we make this last line its own subexpression to be explicit.

```suggestion
		(c.config.ExcludePersonalRepositories && r.IsPersonalRepository())
```